src/RoslynInsertionTool/RoslynInsertionTool/VersionsUpdater.cs

```diff
@@ -67,11 +67,7 @@ public void UpdateComponentVersion(string assemblyName, Version newVersion)
             //   values:                     const string AssemblyNameVersion = "<version-number>";
             //   footer: closing tag:    #>
             var lines = content.Split('\n').Select(line => line.TrimEnd('\r')).ToList();
-            var finalLines = new List<string>();
             bool IsVersionLine(string line) => line.TrimStart().StartsWith("const");
-            var headerLines = lines.TakeWhile(line => !IsVersionLine(line)).ToList();
-            var valueLines = lines.Skip(headerLines.Count).TakeWhile(IsVersionLine).ToList();
```

Comment:
Instead of trying to get the group of the version lines (by splitting on new lines), I just go through all the lines, check if its a version line, and update or insert based on the order in the file.  This seems more resilient to random comments or new lines in between versions.